src/libtls/tests/suites/test_hkdf.c

```diff
@@ -0,0 +1,429 @@
+		0x02, 0x00, 0x00, 0x76, 0x03, 0x03, 0x70, 0x71,
+		0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79,
+		0x7a, 0x7b, 0x7c, 0x7d, 0x7e, 0x7f, 0x80, 0x81,
+		0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89,
+		0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x20, 0xe0,
+		0xe1, 0xe2, 0xe3, 0xe4, 0xe5, 0xe6, 0xe7, 0xe8,
+		0xe9, 0xea, 0xeb, 0xec, 0xed, 0xee, 0xef, 0xf0,
+		0xf1, 0xf2, 0xf3, 0xf4, 0xf5, 0xf6, 0xf7, 0xf8,
+		0xf9, 0xfa, 0xfb, 0xfc, 0xfd, 0xfe, 0xff, 0x13,
+		0x01, 0x00, 0x00, 0x2e, 0x00, 0x33, 0x00, 0x24,
+		0x00, 0x1d, 0x00, 0x20, 0x9f, 0xd7, 0xad, 0x6d,
+		0xcf, 0xf4, 0x29, 0x8d, 0xd3, 0xf9, 0x6d, 0x5b,
+		0x1b, 0x2a, 0xf9, 0x10, 0xa0, 0x53, 0x5b, 0x14,
+		0x88, 0xd7, 0xf8, 0xfa, 0xbb, 0x34, 0x9a, 0x98,
+		0x28, 0x80, 0xb6, 0x15, 0x00, 0x2b, 0x00, 0x02,
+		0x03, 0x04,
+	);
+
+	chunk_t ecdhe = chunk_from_chars(
+		/* df4a291baa1eb7cfa6934b29b474baad2697e29f1f920dcc77c8a0a088447624 */
+		0xdf, 0x4a, 0x29, 0x1b, 0xaa, 0x1e, 0xb7, 0xcf,
+		0xa6, 0x93, 0x4b, 0x29, 0xb4, 0x74, 0xba, 0xad,
+		0x26, 0x97, 0xe2, 0x9f, 0x1f, 0x92, 0x0d, 0xcc,
+		0x77, 0xc8, 0xa0, 0xa0, 0x88, 0x44, 0x76, 0x24,
+	);
+
+	chunk_t expected_client_handshake_traffic_secret = chunk_from_chars(
+		/* ff0e5b965291c608c1e8cd267eefc0afcc5e98a2786373f0db47b04786d72aea */
+		0xff, 0x0e, 0x5b, 0x96, 0x52, 0x91, 0xc6, 0x08,
+		0xc1, 0xe8, 0xcd, 0x26, 0x7e, 0xef, 0xc0, 0xaf,
+		0xcc, 0x5e, 0x98, 0xa2, 0x78, 0x63, 0x73, 0xf0,
+		0xdb, 0x47, 0xb0, 0x47, 0x86, 0xd7, 0x2a, 0xea,
+	);
+
+	chunk_t expected_server_handshake_traffic_secret = chunk_from_chars(
+		/* a2067265e7f0652a923d5d72ab0467c46132eeb968b6a32d311c805868548814 */
+		0xa2, 0x06, 0x72, 0x65, 0xe7, 0xf0, 0x65, 0x2a,
+		0x92, 0x3d, 0x5d, 0x72, 0xab, 0x04, 0x67, 0xc4,
+		0x61, 0x32, 0xee, 0xb9, 0x68, 0xb6, 0xa3, 0x2d,
+		0x31, 0x1c, 0x80, 0x58, 0x68, 0x54, 0x88, 0x14,
+	);
+
+	chunk_t expected_client_handshake_key = chunk_from_chars(
+		/* 7154f314e6be7dc008df2c832baa1d39 */
+		0x71, 0x54, 0xf3, 0x14, 0xe6, 0xbe, 0x7d, 0xc0,
+		0x08, 0xdf, 0x2c, 0x83, 0x2b, 0xaa, 0x1d, 0x39,
+	);
+
+	chunk_t expected_client_handshake_iv = chunk_from_chars(
+		/* 71abc2cae4c699d47c600268 */
+		0x71, 0xab, 0xc2, 0xca, 0xe4, 0xc6, 0x99, 0xd4,
+		0x7c, 0x60, 0x02, 0x68,
+	);
+
+	chunk_t expected_server_handshake_key = chunk_from_chars(
+		/* 844780a7acad9f980fa25c114e43402a */
+		0x84, 0x47, 0x80, 0xa7, 0xac, 0xad, 0x9f, 0x98,
+		0x0f, 0xa2, 0x5c, 0x11, 0x4e, 0x43, 0x40, 0x2a,
+	);
+
+	chunk_t expected_server_handshake_iv = chunk_from_chars(
+		/* 4c042ddc120a38d1417fc815 */
+		0x4c, 0x04, 0x2d, 0xdc, 0x12, 0x0a, 0x38, 0xd1,
+		0x41, 0x7f, 0xc8, 0x15,
+	);
+
+	chunk_t c_secret, c_key, c_iv, s_secret, s_key, s_iv;
+
+	tls_hkdf_t *hkdf = tls_hkdf_create(HASH_SHA256, NULL);
+
+	ck_assert(hkdf->set_shared_secret(hkdf, &ecdhe));
+
+	/* Generate client handshake traffic secret */
+	ck_assert(hkdf->generate_secret(hkdf, TLS_HKDF_C_HS_TRAFFIC, &handshake, &c_secret));
+	ck_assert(chunk_equals(expected_client_handshake_traffic_secret, c_secret));
```

Comment:
Use the following instead as it prints the contents if the chunks are not equal:
```suggestion
	ck_assert_chunk_eq(expected_client_handshake_traffic_secret, c_secret);
```